src/index.js

```diff
@@ -91,6 +91,11 @@ export default class VueRouter {
 
     // main app already initialized.
     if (this.app) {
+      app.$once('hook:destroyed', () => {
```

Comment:
if you add it inside the if, creating an app and then destroying the main app after it would still result in a memory leak, right?

Could you add some tests to api.spec.js or another new file checking the apps array is emptied correctly, please?

Reply:
The main app, would typically be destroyed last (I would think)

The reason for placing it inside of the if would be that it would t destroy the main app, only apps that came along after the main app was instantiated.

Reply:
I'll see what I can do a bout adding a few tests.

Reply:
Another question would be, what if the main app is destroyed  y the other secondary apps are still alive? Should the main app be removed as well, and should the next app in the apps array (with the lowest index number) be copied to `this.app` ?